packages/fiori-elements-writer/src/data/manifestSettings.ts

```diff
@@ -34,6 +36,17 @@ export function extendManifestJson<T>(
         ].includes(feApp.template.type)
     ) {
         templatePath = TemplateType.ListReportObjectPage;
+
+        // starting with UI5 v1.94.0, contextPath should be used instead of 'entitySet' in manifest for v4 LROP based apps
+        const minVersion = semVer.coerce(feApp.ui5?.minUI5Version);
+        if (!minVersion || semVer.gte(minVersion, '1.94.0')) {
+            const entityConfig = (feApp.template.settings as LROPSettings).entityConfig as ManifestEntitySettings;
+            entityConfig.contextPath = `/${entityConfig.mainEntityName}`;
+
+            if (entityConfig.navigationEntity?.EntitySet) {
+                entityConfig.navigationEntity.contextPath = `${entityConfig.contextPath}/${entityConfig.navigationEntity.Name}`;
```

Comment:
For the first object page this is not correct. Both LR and OP share the same entitySet, there is no real navigation between them. Thus `${entityConfig.contextPath}` is sufficient. Only follow-up pages have to consider the navigationEntity. Example:

 ```
"targets": {
                "TravelList": {
                    "type": "Component",
                    "id": "TravelList",
                    "name": "sap.fe.templates.ListReport",
                    "options": {
                        "settings": {
                            "contextPath": "/Travel",
                            "navigation": {
                                "Travel": {
                                    "detail": {
                                        "route": "TravelObjectPage"
                                    }
                                }
                            }
                        }
                    }
                },
                "TravelObjectPage": {
                    "type": "Component",
                    "id": "TravelObjectPage",
                    "name": "sap.fe.templates.ObjectPage",
                    "options": {
                        "settings": {
                            "contextPath": "/Travel",
                            "navigation": {
                                "_Booking": {
                                    "detail": {
                                        "route": "Travel__BookingObjectPage"
                                    }
                                }
                            }
                        }
                    }
                },
                "Travel__BookingObjectPage": {
                    "type": "Component",
                    "id": "Travel__BookingObjectPage",
                    "name": "sap.fe.templates.ObjectPage",
                    "options": {
                        "settings": {
                            "contextPath": "/Travel/_Booking",
                            "navigation": {}
                        }
                    }
                }
            }
        },


```

Reply:
@schreckstefan Do you see something different? I think this is what has been implemented.

Reply:
OK, this looks great. Then I was misunderstanding your coding